packages/services/rate-limit/src/api.ts

```diff
@@ -1,50 +1,166 @@
-import { z } from 'zod';
-import { handleTRPCError } from '@hive/service-common';
-import type { FastifyRequest } from '@hive/service-common';
-import type { inferRouterInputs, inferRouterOutputs } from '@trpc/server';
-import { initTRPC } from '@trpc/server';
+import { FastifyLoggerInstance, reportReadiness } from '@hive/service-common';
+import * as Sentry from '@sentry/node';
+import {
+  createRouter,
+  FromSchema,
+  Response,
+  RouterInput,
+  RouterOutput,
+  useErrorHandling,
+  useOpenAPI,
+} from '@whatwg-node/router';
 import type { Limiter } from './limiter';
 
 export interface Context {
-  req: FastifyRequest;
   limiter: Limiter;
 }
 
-const t = initTRPC.context<Context>().create();
-const errorMiddleware = t.middleware(handleTRPCError);
-const procedure = t.procedure.use(errorMiddleware);
+const VALIDATION = {
+  type: 'object',
+  properties: {
+    id: {
+      type: 'string',
+      minLength: 1,
+    },
+    entityType: {
+      type: 'string',
+      enum: ['organization', 'target'],
+    },
+    type: {
+      type: 'string',
+      enum: ['operations-reporting'],
+    },
+    token: {
+      type: 'string',
+      nullable: true,
+    },
+  },
+  additionalProperties: false,
+  required: ['id', 'entityType', 'type', 'token'],
+} as const;
 
-export type RateLimitInput = z.infer<typeof VALIDATION>;
+export type RateLimitInput = FromSchema<typeof VALIDATION>;
 
-const VALIDATION = z
```

Comment:
to make the diff smaller, we can actually use the same zod schema and use https://github.com/StefanTerdell/zod-to-json-schema ?
just to show the actual difference between tRPC and this instead of also json-schema<->zod change?

Reply:
But it should be a constant type definition. I don't think that library provides that.

Reply:
We can also implement a helper lib like Zod.

Reply:
Changed to Zod